### PR TITLE
dynamic priority fee

### PR DESCRIPTION
**Pull request by enriquesouza**
https://github.com/voltix-vault/voltix/issues/224

I've implemented the dynamic priority fees using the Solana's RPC, therefore the networks has issues

Let's see if it gets sorted until April's 15h.

https://cointelegraph.com/news/solana-devs-target-mid-april-to-fix-failed-transaction-dilemma

image

### CODE REVIEW

File: voltixApp/VoltixApp/Services/Solana/SolanaService.swift [hunk 1]
```diff
@@ -91,7 +91,54 @@ class SolanaService {
...
+        
+        let data = try await postRequest(with: requestBody)
+        let decoder = JSONDecoder()
+        let response = try decoder.decode(PrioritizationFeeResponse.self, from: data)
+        
+        let fees = response.result.map { $0.prioritizationFee }
+        let averageFeeIncludingZeros = fees.reduce(0, +) / fees.count
+        
+        let nonZeroFees = fees.filter { $0 > 0 }
+        let averageFeeExcludingZeros = nonZeroFees.isEmpty ? 0 : nonZeroFees.reduce(0, +) / nonZeroFees.count
```
Comment (johnnyluo) [inline voltixApp/VoltixApp/Services/Solana/SolanaService.swift:123]: As far as I understand , the only thing we need is the highPriorityFee , why we `averageFeeIncludingZeros` `averageFeeExcludingZeros` , `medianFee` ?

Comment (enriquesouza) [reply voltixApp/VoltixApp/Services/Solana/SolanaService.swift:123]: Yes, at the moment we are only using the high one.

Comment (johnnyluo) [reply voltixApp/VoltixApp/Services/Solana/SolanaService.swift:123]: do you want to remove it , add when we need it?

Comment (enriquesouza) [reply voltixApp/VoltixApp/Services/Solana/SolanaService.swift:123]: Sure. I am doing it.

**Review by johnnyluo — APPROVED** [review]
LGTM

**Pull request merged**